src/Kmd.Logic.Gateway.Automation/PublishFile/PublishFileModel.cs

```diff
@@ -4,8 +4,20 @@ namespace Kmd.Logic.Gateway.Automation.PublishFile
 {
     internal class PublishFileModel
     {
-        public IEnumerable<Product> Products { get; set; }
+        private IEnumerable<Product> products;
 
-        public IEnumerable<Api> Apis { get; set; }
+        public IEnumerable<Product> Products
```

Comment:
You can always do `public IEnumerable<Product> Products { get; set; } = new List<Product>()` (and same with Apis). If in YAML file no such entities are provided, then it will be empty list.

Reply:
That was the way I implemented it originally. With the present version of code, it works too. But still we have a catch that it doesn't stop someone from assigning it a null tomorrow and thus breaking the working code.

Reply:
I do not think that we should bother much about it. I would rather remove this boilerplate, for the cleaner code's sake. What do you think guys? @Araballi @NakulSeerpu @mlatoszek ?

Reply:
Here is an update. I just tested with the initializers that @jaceks2106  suggested. I see that the properties are overwritten as NULL as we read from the file(Not sure why I felt it worked before). 

Here is what I had in the yaml file

![image](https://user-images.githubusercontent.com/12545474/99931545-eec3c700-2d7a-11eb-9b7d-6c8aec8ecca1.png)



Hence we have only two choices.

- Either retain the way I have coded it
- Do null check in all the places where they are referenced

Reply:
**An update to this case**

Tested the initializer syntax as below
![image](https://user-images.githubusercontent.com/12545474/99932216-6abf0e80-2d7d-11eb-95c8-742d7f806d22.png)
Here is what I had in the yaml
![image](https://user-images.githubusercontent.com/12545474/99932167-4bc07c80-2d7d-11eb-96b1-6918644ab61d.png)

and it broke with **null** reference exception. 

Following are the ways we may deal with this

- Retain the way I have coded it
- Add the initializer(this works if the file has elements like Products:[] and Apis:: [] OR if the property itself is not specified) and do a NULL check immediately after reading from the file and finding it null(to deal with the above case shown in the screenshot)
- Do null check everywhere

For me, I prefer the way I have done it since it is a centralized way of preventing the error. Let me know what the team prefers

Reply:
Instead of having null check everyhwere, its better as the way it is. My opinion.

Reply:
Ok. So let's close this comment. I have just tested this, and my idea really does not work. That's weird.....